test/cases/ref.ts

```diff
@@ -0,0 +1,40 @@
+export var schema = 
+{
+  "title": "Referencing",
+  "type": "object",
+  "properties": {
+    "foo": {
+      "$ref": "test/resources/ReferencedType.json"
+    }
+  },
+  "required": ["foo"],
+  "additionalProperties": false
+}
+
+export var configurations = [
+  {
+    settings: {
+      declareReferenced: true
```

Comment:
I think a batch mode would be nice for sure.  Or even a grunt and/or gulp plugin.  

And I did not realize that interfaces are [merged](https://www.typescriptlang.org/docs/handbook/declaration-merging.html).  But I still think for readability's sake we should cut down on clutter... 

I'm inclined to ask if we can leave this for now (since the option already exists) and take this batch mode as a second pass of the work.  Do you think that's possible?